packages/tools/replay-tool/src/replayMessages.ts

```diff
@@ -757,19 +782,37 @@ export class ReplayTool {
             return false;
         }
 
-        if (!sameContent(content, content2)) {
+        // Check if the two snapshots match.
+        let failed = true;
+        let error: any;
+        if (content.op === content2.op) {
+            // Deep compare the normalized snapshots. If they do not match, catch the error and display it.
+            try {
+                strict.deepStrictEqual(content.normalizedSnapshot, content2.normalizedSnapshot);
+                failed = false;
+            } catch (e) {
+                error = e;
+            }
+        }
+
+        if (failed) {
             // eslint-disable-next-line max-len
-            this.reportError(`\nOp ${op}: Discrepancy between ${name1} & ${name2}! Likely a bug in snapshot load-save sequence!`);
-            fs.mkdirSync(dir, { recursive: true });
+            this.reportError(`\nOp ${op}: Discrepancy between ${name1} & ${name2}! Likely a bug in snapshot load-save sequence!`, error);
+
+            // Write the failed snapshots under 'FailedSnapshot' sub-directory of the current directory. This will in
+            // debugging by looking into the complete snapshot.
+            const failedDir = `${dir}/FailedSnapshots`;
+            fs.mkdirSync(failedDir, { recursive: true });
 
-            this.expandForReadabilityAndWriteOut(content, `${dir}/${name1}`);
-            this.expandForReadabilityAndWriteOut(content2, `${dir}/${name2}`);
+            this.expandForReadabilityAndWriteOut(content, `${failedDir}/${name1}`);
+            this.expandForReadabilityAndWriteOut(content2, `${failedDir}/${name2}`);
 
             if (this.args.windiff) {
-                console.log(`windiff.exe "${dir}/${name1}_expanded.json" "${dir}/${name2}_expanded.json"`);
```

Comment:
This needs to change for a couple of reasons:
1. The snapshot files written to disk is not normalized. So, windiff here won't accurately show the diffs. We could write normalized files to disk to get around this, but I think that should be in original format as generated by the Container.
2. deepStrictEqual outputs a decent diff of where the failure happened.
3. windiff wont work in linux / mac.